addons/web/static/src/js/chrome/action_manager.js

```diff
@@ -388,21 +388,27 @@ var ActionManager = Widget.extend({
         }
 
         return this._startController(controller).then(function (controller) {
+            var prevDialogOnClose;
             if (self.currentDialogController) {
+                if (action.target === 'new') {
+                    prevDialogOnClose = self.currentDialogController.dialog.transmittableOnClose;
+                }
                 self._closeDialog(true);
             }
 
+            var transmittableOnClose = prevDialogOnClose || options.on_close;
             var dialog = new Dialog(self, _.defaults({}, options, {
                 buttons: [],
                 dialogClass: controller.className,
                 title: action.name,
                 size: action.context.dialog_size,
             }));
+            dialog.transmittableOnClose = transmittableOnClose;
```

Comment:
i don't like this, i'd store it on the currentDialogController, like
self.currentDialogController.onClose = prevDialogOnClose || options.on_close;